tee-worker/litentry/core/assertion-build/src/a14.rs

```diff
@@ -114,26 +51,28 @@ pub fn build(
 		}
 	}
 	let mut value = false;
-	let mut client = A14Client::new(data_provider_config);
-
-	for address in polkadot_addresses {
-		let data = A14Data {
-			params: A14DataParams { address },
-			include_metadata: false,
-			include_widgets: false,
-		};
-		let response = client.send_request(&data)?;
-
-		let result = response
-			.data
-			.get("result")
-			.and_then(|r| r.as_bool())
-			.ok_or(Error::RequestVCFailed(Assertion::A14, ErrorDetail::ParseError))?;
-		if result {
-			value = result;
-			break
-		}
-	}
+	let mut client = AchainableClient::new(data_provider_config);
+
+	loop_with_abort_strategy(
+		polkadot_addresses,
+		|address| {
+			let data = LabelQueryReq {
+				params: LabelQueryReqParams { address: address.clone() },
+				include_metadata: false,
+				include_widgets: false,
+			};
+			let result = client
+				.query_label("a719e99c-1f9b-432e-8f1d-cb3de0f14dde", &data)
+				.map_err(|e| Error::RequestVCFailed(Assertion::A14, e.into_error_detail()))?;
+			if result {
+				value = result;
+				Ok(LoopControls::Break)
```

Comment:
Seems like `LoopControls` will only ever have two variants, In that case can we use `Option<>` ? I think that's more idiomatic, wdyt?

Reply:
IMO LoopControls is better than using Option<>
1 LoopControls is more readable
2 LoopControls is more extensible, e.g. we can add a new value: LoopControls.RETURN